### PR TITLE
BRO 9 cards

### PR DESCRIPTION
**Pull request by Simisays**

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/mishra_excavation_prodigy.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+Name:Mishra, Excavation Prodigy
+ManaCost:2 R
+Types:Legendary Creature Human Artificer
+PT:2/1
+K:Haste
+A:AB$ Draw | Cost$ 1 T | SubAbility$ DBDiscard | SpellDescription$ Draw a card, then discard a card.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/mishra_excavation_prodigy.txt:6]: Discard is cost

File: forge-gui/res/cardsfolder/upcoming/urza_prince_of_kroog.txt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Name:Urza, Prince of Kroog
+ManaCost:2 W U
+Types:Legendary Creature Human Artificer
+PT:2/3
+S:Mode$ Continuous | Affected$ Creature.Artifact+YouCtrl | AddPower$ 2 | AddToughness$ 2 | Description$ Other artifact creatures you control get +2/+2.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/urza_prince_of_kroog.txt:5]: Wrong Description

File: forge-gui/res/cardsfolder/upcoming/mishra_excavation_prodigy.txt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+Name:Mishra, Excavation Prodigy
+ManaCost:2 R
+Types:Legendary Creature Human Artificer
+PT:2/1
+K:Haste
+A:AB$ Draw | Cost$ 1 T Discard<1/Card> | SpellDescription$ Draw a card.
+T:Mode$ DiscardedAll | ValidPlayer$ You | ValidCard$ Card.Artifact| TriggerZones$ Battlefield | ActivationLimit$ 1 | Execute$ TrigMana | TriggerDescription$ Whenever you discard one or more artifact cards, add {R}{R}. This ability triggers only once each turn.
+SVar:TrigMana:DB$ Mana | Produced$ R R
+DeckHints:Type$Artifact
+DeckHas:Ability$Discard
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/mishra_excavation_prodigy.txt:10]: There is a way to get the AI to prefer discarding artifacts when this is in play

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/mishra_excavation_prodigy.txt:10]: ah AILogic$ DiscardLands right? does this should work with the urza one as well right ?

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/mishra_excavation_prodigy.txt:10]: Oh nvm it's SVar:AIPreference:DiscardCost$Card.Artifact right

File: forge-gui/res/cardsfolder/upcoming/feldon_ronom_excavator.txt [hunk 4]
```diff
@@ -0,0 +1,15 @@
+Name:Feldon, Ronom Excavator
+ManaCost:1 R
+Types:Legendary Creature Human Artificer
+PT:2/2
+K:Haste
+K:CARDNAME can't block.
+T:Mode$ DamageDoneOnce | ValidTarget$ Card.Self | TriggerZones$ Battlefield | Execute$ TrigExile |  TriggerDescription$ Whenever NICKNAME is dealt damage, exile that many cards from the top of your library. Choose one of them. Until the end of your next turn, you may play that card.
+SVar:TrigExile:DB$ Dig | Defined$ You | DigNum$ X | ChangeNum$ All | DestinationZone$ Exile | RememberChanged$ True | SubAbility$ DBChoose
+SVar:DBChoose:DB$ ChooseCard | Choices$ Card.IsRemembered | ChoiceZone$ Exile | Defined$ You | Amount$ 1 | Mandatory$ True | SubAbility$ DBEffect
+SVar:DBEffect:DB$ Effect | RememberObjects$ ChosenCard | StaticAbilities$ Play | SubAbility$ DBCleanup
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/feldon_ronom_excavator.txt:10]: Missing Duration$

Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/feldon_ronom_excavator.txt:10]: | ForgetOnMoved$ Exile

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/feldon_ronom_excavator.txt:10]: Oh until your next turn, missed that

File: forge-gui/res/cardsfolder/upcoming/mishra_excavation_prodigy.txt (hunk 3 above)
Comment (Northmoc) [reply forge-gui/res/cardsfolder/upcoming/mishra_excavation_prodigy.txt:10]: Right, like Fleshgrafter

File: forge-gui/res/cardsfolder/upcoming/splitting_the_powerstone.txt [hunk 5]
```diff
@@ -0,0 +1,8 @@
+Name:Splitting the Powerstone
+ManaCost:2 U
+Types:Sorcery
+A:SP$ Token | Cost$ 2 U Sac<1/Artifact> | TokenAmount$ 2 | TokenTapped$ True | TokenScript$ c_a_powerstone | TokenOwner$ You | SubAbility$ DBDraw | SpellDescription$ As an additional cost to cast this spell, sacrifice an artifact.Create two tapped Powerstone tokens. If the sacrificed artifact was legendary, draw a card. (The tokens are artifacts with "{T}: Add {C}. this mana can't be spent to cast a nonartifact spell.")
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/splitting_the_powerstone.txt:4]: SpellDescription first part should already be auto built?

File: forge-gui/res/cardsfolder/upcoming/hurkyl_master_wizard.txt [hunk 6]
```diff
@@ -0,0 +1,24 @@
+Name:Hurkyl, Master Wizard
+ManaCost:1 U U
+Types:Legendary Creature - Human Wizard Advisor
+PT:2/4
+T:Mode$ Phase | Phase$ End of Turn | ValidPlayer$ You | TriggerZones$ Battlefield | CheckSVar$ X | SVarCompare$ GE1 | Execute$ TrigDig | TriggerDescription$ At the beginning of your end step, if you've cast a noncreature spell this turn, reveal the top five cards of your library. For each card type among noncreature spells you've cast this turn, you may put a card of that type from among the revealed cards into your hand. Put the rest on the bottom of your library in a random order.
+SVar:TrigDig:DB$ Dig | DigNum$ 5 | Reveal$ True | NoMove$ True | RememberRevealed$ True | SubAbility$ DBChooseCardOne
+SVar:DBChooseCardOne:DB$ ChooseCard | Defined$ You | Choices$  Instant.YouCtrl+IsRemembered  | ImprintChosen$ True | ConditionCheckSVar$ CheckInstant | ConditionSVarCompare$ EQ1 | ChoiceTitle$ You may choose an instant card if you cast an instant spell this turn | ChoiceZone$ Library | AILogic$ BestCard | SubAbility$ DBChooseCardTwo
```
Comment (Hanmac) [inline forge-gui/res/cardsfolder/upcoming/hurkyl_master_wizard.txt:7]: the choices should be optional?

File: forge-gui/res/cardsfolder/upcoming/mishra_excavation_prodigy.txt [hunk 7]
```diff
@@ -0,0 +1,12 @@
+Name:Mishra, Excavation Prodigy
+ManaCost:2 R
+Types:Legendary Creature Human Artificer
+PT:2/1
+K:Haste
+A:AB$ Draw | Cost$ 1 T Discard<1/Card> | SpellDescription$ Draw a card.
+T:Mode$ DiscardedAll | ValidPlayer$ You | ValidCard$ Card.Artifact| TriggerZones$ Battlefield | ActivationLimit$ 1 | Execute$ TrigMana | TriggerDescription$ Whenever you discard one or more artifact cards, add {R}{R}. This ability triggers only once each turn.
```
Comment (Hanmac) [inline forge-gui/res/cardsfolder/upcoming/mishra_excavation_prodigy.txt:7]: space after Artifact before "|"

File: forge-gui/res/cardsfolder/upcoming/hurkyl_master_wizard.txt (hunk 6 above)
Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/hurkyl_master_wizard.txt:7]: I think with Choosecard the default is optional example 
image

File: forge-gui/res/cardsfolder/upcoming/ashnods_harvester.txt [hunk 8]
```diff
@@ -0,0 +1,9 @@
+Name:Ashnod's Harvester
+ManaCost:2
+Types:Artifact Creature Construct
+PT:3/1
+T:Mode$ Attacks | ValidCard$ Card.Self | Execute$ TrigExile | TriggerDescription$ Whenever CARDNAME attacks, exile a card from a graveyard.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/ashnods_harvester.txt:5]: TriggerDescription incorrect

**Review by tool4ever — APPROVED** [review]

**Pull request merged**